src/Language/PureScript/Ide/CaseSplit.hs

```diff
@@ -119,11 +120,17 @@ addClause s wca =
         " = ?" <> (T.strip . T.pack . runIdent $ fName)
   in [s, template]
 
-parseType' :: String -> Type
-parseType' s = let (Right t) = do
-                     ts <- lex "" s
-                     runTokenParser "" (parseType <* P.eof) ts
-               in t
+parseType' :: (MonadError PscIdeError m) =>
+              String -> m Type
+parseType' s = let
+                 parseResult = do
+                   ts <- lex "" s
+                   runTokenParser "" (parseType <* P.eof) ts
+               in
+                 case parseResult of
```

Comment:
Sure, I usually use a placeholder in those cases. It's just to help debugging, so I'd normally put something like `<psc-ide>` in there.

Doesn't really matter though.